test/terra/backends/test_qasm_simulator_matrix_product_state.py

```diff
@@ -83,7 +82,6 @@ class TestQasmMatrixProductStateSimulator(
         QasmSnapshotStatevectorTests,
         QasmSnapshotDensityMatrixTests,
         QasmSnapshotProbabilitiesTests,
-        QasmSnapshotStabilizerTests,
```

Comment:
Why? MPS is supposed to run well on these tests

Reply:
I was getting multiple warnings that MPS does not support save_stabilizer. I did not check what exactly is giving these warnings. It is possible that some of the tests pass and only some give the warning. I can remove from this PR and check more carefully in a separate PR.

Reply:
Do you see these warnings also when you run the tests in the main branch?
- If yes (my guess): then it's a good idea to check it, and this can be done independently of this PR (either before or after this PR is merged), but there's no reason to remove it from this PR.
- If no, then it's something that's specific to this PR, and so has to be investigated before this PR is merged.

I expect the same warnings to appear also for the statevector tests, unless it has already been removed there.

Reply:
@yaelbh , I made a changes following your comments:
1. I put back the stabilizer tests in mps. I will check this in a separate issue/PR.
2. I put back the sort in `reset`.
3. I added a few comments regarding the sort. `apply_measure_internal` requires the qubits to be sorted by the caller function. I prefer not to put a validity check in the method for performance considerations. So instead, I put comments in all relevant places.